framework/src/partitioner/PetscExternalPartitioner.C

```diff
@@ -93,6 +93,15 @@ PetscExternalPartitioner::partition(MeshBase & mesh, const unsigned int n_parts)
   if (mesh.is_replicated() && n_parts > 1)
     preLinearPartition(mesh);
 
+  if ((mesh.n_elem() / n_parts < 28) && _part_package == "parmetis")
```

Comment:
Should we respect users' choice here, i.e. when the package is set by users, we do not do this auto switch?

Reply:
PR is up to do that